File: leetcode-submissions/349_Intersection_of_Two_Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
      vector<int>ans;
      unordered_map<int,int>mp;
      for(int i=0;i<nums2.size();i++){
        mp[nums2[i]]++;
      }
      set<int>s;
      for(int i=0;i<nums1.size();i++){
          if(mp.count(nums1[i])>0){
             s.insert(nums1[i]);
          }
      }
      
      
    for(int ele:s){
        ans.push_back(ele);
    }
    return ans;
    }
};

File: leetcode-submissions/349_Intersection_of_Two_Arrays.ts
function intersection(nums1: number[], nums2: number[]): number[] {
    const map:Map<number , number >= new Map();
    const st:Set<number> = new Set(nums2);
    for(const num of nums1){
        map.set(num , (map.get(num) || 0 )+1);
        
    }
    const ans : number[] = [];
    for(const num of st){
        if(map.has(num)){
            ans.push(num);
        }
    }
    return ans  ;
};